### PR TITLE
update rust toolchain version

### PR DESCRIPTION
**Pull request by fluidvanadium**
new rust drop! i followed suggestions from rustc, clippy, and fmt

### CODE REVIEW

File: zingo-memo/src/lib.rs [hunk 1]
```diff
@@ -33,7 +33,7 @@ pub fn create_wallet_internal_memo_version_0(uas: &[UnifiedAddress]) -> io::Resu
     let mut uas_bytes_vec = Vec::new();
     CompactSize::write(&mut uas_bytes_vec, 0usize)?;
     Vector::write(&mut uas_bytes_vec, uas, |w, ua| {
-        write_unified_address_to_raw_encoding(ua, w)
+        write_unified_address_to_raw_encoding(ua, w.clone())
```
Comment (zancas) [inline zingo-memo/src/lib.rs:36]: Does this change have something to do with the Rust drop?

Comment (fluidvanadium) [reply zingo-memo/src/lib.rs:36]: yes

Comment (fluidvanadium) [reply zingo-memo/src/lib.rs:36]: all this PR does is follow compiler instructions

**Comment by zancas** [comment]
Failing CI.

**Comment by zancas** [comment]
I am still seeing CI failures on this.

**Comment by fluidvanadium** [comment]
these screenshots show that this update changes how value transfers are understood. the value transfer data transformation is... arcane

before

after

File: zingo-memo/src/lib.rs [hunk 2]
```diff
@@ -65,34 +65,36 @@ pub fn parse_zingo_memo(memo: [u8; 511]) -> io::Result<ParsedMemo> {
...
         let (typecode, data): (u32, &[u8]) = match receiver {
             Receiver::Orchard(ref data) => (3, data),
             Receiver::Sapling(ref data) => (2, data),
             Receiver::P2pkh(ref data) => (0, data),
             Receiver::P2sh(ref data) => (1, data),
             Receiver::Unknown { typecode, ref data } => (*typecode, data.as_slice()),
         };
-        CompactSize::write(&mut w, typecode as usize)?;
-        CompactSize::write(&mut w, data.len())?;
+        CompactSize::write(w.clone(), typecode as usize)?;
```
Comment (Oscar-Pepper) [inline zingo-memo/src/lib.rs:83]: i will need to do some research on these rust changes but cloning readers / writers doesnt sit right with me....

**Comment by zancas** [comment]
These test failures could be explained by this known bug in clippy 1.79:

https://github.com/rust-lang/rust-clippy/issues/13170

**Pull request closed without merging**